### ISSUE TITLE
Improve documentation by integrating VuePress

### ISSUE THREAD

**Issue by kasvith** [knex/knex#5915]
Often when using `Knex.js` documentation I find it rather difficult to search for something. Also, the font size is too small. If we moved to a system like VuePress we get a lot of things out of the box

- Search(default or Algolia)
- Rendering support with themes(dark mode included)
- Ability to run JS(for example we can run Knex snippets as we already doing)
- Documentation with Markdown
- And ton more

[Objection.js](https://vincit.github.io/objection.js/) is already using VuePress and I find it very easy to navigate and use.

By doing this I hope Knex documentation will be much accessible and pleasant to work with.

**Comment by rubengmurray** [comment]
This can be moved to knex/documentation repo I would've thought @mercmobily

**Comment by mercmobily** [comment]
No, the documentation repo is legacy and must be archived. I made the mistake of adding an issue there, and will need to move it over here.

**Comment by mercmobily** [comment]
Thanks for proposing this improvement! The migration to VuePress would indeed modernise our documentation and provide a better user experience.

Looking at the discussion, @vencho-mdp started working on this in a separate repository (https://github.com/vencho-mdp/knexjs-docs-vuepress) but the effort appears to have stalled. The documentation has since been moved from the separate `knex/documentation` repository directly into the main Knex repository, which changes how we'd approach this migration.

**Benefits of VuePress**

Your points about VuePress are spot on:
- Built-in search functionality (with Algolia integration option)
- Better typography and readability
- Dark mode support
- Markdown-based documentation (easier to maintain)
- Better mobile responsiveness
- Sidebar navigation with collapsible sections

The Objection.js documentation is indeed a good example of VuePress done well.

**Implementation Considerations**

Since the documentation now lives in the main repository, the migration would involve:

1. Setting up VuePress in the main Knex repo
2. Converting existing documentation from the current React/Webpack setup to Markdown/VuePress components
3. Preserving existing URLs to avoid breaking external links (critical for backward compatibility)
4. Migrating the dynamic changelog component
5. Setting up the build pipeline and deployment

**Next Steps**

If you're still interested in working on this:

1. Create a proof-of-concept branch with VuePress set up
2. Migrate a few key documentation pages as examples
3. Ensure URL compatibility with redirects where needed
4. Set up the changelog integration (this was identified as the most complex part)

The main challenge will be maintaining backward compatibility with existing documentation URLs, as many projects and tutorials link to specific sections of the Knex documentation.

Would you like to take this on? We'd be happy to review a PR that implements this migration. The key requirements would be:

- Preserve existing URL structure or provide redirects
- Maintain all current documentation content
- Ensure the changelog remains functional
- Set up proper build and deployment scripts

**Comment by mercmobily** [comment]
The documentation has actually already been migrated from the old React/Webpack setup to **VitePress** (version 0.22.4), not VuePress. VitePress is the successor to VuePress, built on Vite for faster builds and better performance.

This migration was completed in PR #5121 (which was later moved to the documentation repository as PR knex/documentation#400) and then integrated into the main knex repository in commit a58208cd (#5792). The documentation now lives in the `docs/` folder of the main repository and uses VitePress with its modern build system.

**Current status:**
- Documentation is already using VitePress
- The docs are in the main repository under `/docs`
- Build commands are `npm run dev`, `npm run build`, and `npm run serve` in the docs folder
- Deployment workflow is set up via GitHub Actions

So this issue can be closed as the migration has been successfully completed. The documentation is now more maintainable and performant with VitePress.

**Related issues:**
- #5923 - Integrate Vitepress (closed, completed)
- #5944 - Outdated changelog on website (addressed in migration)
